spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -470,17 +470,24 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   /**
    * module predict, return the probability distribution
    * @param dataset dataset for prediction
+   * @param batchSize total batchSize for all partitions.
+   *                  if -1, default is 4 * partitionNumber of datatset
+   * @param repeatMemory whether to use same memory for each batch predict results
    */
-  def predict(dataset: RDD[Sample[T]]): RDD[Activity] = {
-    Predictor(this).predict(dataset)
+  def predict(dataset: RDD[Sample[T]],
+              batchSize: Int = -1,
+              repeatMemory: Boolean = false): RDD[Activity] = {
```

Comment:
why false by default?

Reply:
In our user group, a user uses rdd.take() to get repeated predict results. And he thinks it is confused for him. Other users may have such confuse too. So I use default "false" to get no repeat results.

Reply:
ok - change the name to sth like "shareBuffer"

Reply:
ok